### PR TITLE
collection task done

### PR DESCRIPTION
**Pull request by dimonl**
Collection task to check

### CODE REVIEW

File: collection/collection/collection.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+			return nil
+		}
+		currentEl = currentEl.Next()
+		currentElIndex++
+
+	}
+	return currentEl
+}
+
+func (c *Collection) Remove(index int) {
```
Comment (MalakhovVladislav) [inline collection/collection/collection.go]: you can use your method Get(index int) here to get an element to remove

File: collection/collection/collection.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		el.prev = nil
+	} else {
+		lastPost := c.tail
+		lastPost.next = &el
+		el.prev = c.tail
+		c.tail = &el
+	}
+}
+
+func (c *Collection) Length() int {
```
Comment (MalakhovVladislav) [inline collection/collection/collection.go]: we can add a field "len" to our struct, and do len++ on add and len-- on remove

File: collection/collection/collection.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
+package collection
+
+import "fmt"
+
+type Element struct {
```
Comment (MalakhovVladislav) [inline collection/collection/collection.go]: move it to the other file, please
1 struct - 1 file

File: collection/collection/collection.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+		currentEl = currentEl.Next()
+		currentElIndex++
+
+	}
+	return currentEl
+}
+
+func (c *Collection) Remove(index int) {
+
+	if c.Length() == 1 {
```
[the remark this answers is not in the document]
Comment (dimonl) [reply collection/collection/collection.go]: done. checking index in func

File: collection/collection/collection.go (hunk 1 above)
Comment (dimonl) [reply collection/collection/collection.go]: done, doing refactoring func with using methods of element struct and methods of collections

File: collection/collection/collection.go (hunk 2 above)
Comment (dimonl) [reply collection/collection/collection.go]: adding field length - and refactor functions add and remove

File: collection/collection/collection.go (hunk 3 above)
Comment (dimonl) [reply collection/collection/collection.go]: refactor code - move struct Element in to the other file.

**Pull request merged**